### PR TITLE
share community button #56

### LINKED ISSUE #56: Share community button
**Is your feature request related to a problem? Please describe.**
A button to share a community.

**Describe the solution you'd like**
If it's mobile should use the [share web api](https://web.dev/web-share/) and on desktop the button should be able of copy the link to the community.

### PR DESCRIPTION
**Pull request by koushik-dev**
close #56

### CODE REVIEW

File: src/app/pages/main/components/sidenav/sidenav.component.ts [hunk 1]
```diff
@@ -22,4 +24,21 @@ export class SidenavComponent {
   onCloseInfo() {
     this.closeInfo.emit();
   }
+
+  onShare() {
+    const nav = window.navigator as any;
```
Comment (tonivj5) [inline src/app/pages/main/components/sidenav/sidenav.component.ts:29]: Any reason what `any` is required?

```suggestion
    const nav = window.navigator;
```

File: src/app/pages/main/components/sidenav/sidenav.component.ts [hunk 2]
```diff
@@ -22,4 +24,21 @@ export class SidenavComponent {
...
+          title: this.community.name,
+          url: location.href,
+        })
+        .then(() => console.log('Successful share'))
+        .catch(error => console.log('Error sharing', error));
+    }
+  }
+
+  onCopy() {
+    this.clipboard.copy(location.href);
```
Comment (tonivj5) [inline src/app/pages/main/components/sidenav/sidenav.component.ts:42]: I think we should notify the user, maybe a [snackbar](https://material.angular.io/components/snack-bar/overview) or something so?

File: src/app/pages/main/components/sidenav/sidenav.component.ts [hunk 3]
```diff
@@ -22,4 +24,21 @@ export class SidenavComponent {
...
+  onShare() {
+    const nav = window.navigator as any;
+    if (nav.share) {
+      nav
+        .share({
+          title: this.community.name,
+          url: location.href,
+        })
+        .then(() => console.log('Successful share'))
+        .catch(error => console.log('Error sharing', error));
```
Comment (tonivj5) [inline src/app/pages/main/components/sidenav/sidenav.component.ts:37]:
```suggestion
        .catch(error => console.error('Error sharing', error));
```

**Comment by tonivj5** [comment]
By the way, `window.navigator.share` crash my tab if I use it in desktop :exploding_head:

**Comment by koushik-dev** [comment]
Thanks @tonivj5 !

Added snack bar and keyboard accessibility for the buttons.

**Comment by Caballerog** [comment]
Thanks!

**Pull request merged**